src/main/auth.c

```diff
@@ -554,34 +554,41 @@ int rad_authenticate(REQUEST *request)
 	 *	modules has decided that a proxy should be used. If
 	 *	so, get out of here and send the packet.
 	 */
-	if (
 #ifdef WITH_PROXY
-	    (request->proxy == NULL) &&
+	if (request->proxy == NULL)
 #endif
-	    ((tmp = fr_pair_find_by_num(request->config, PW_PROXY_TO_REALM, 0, TAG_ANY)) != NULL)) {
-		REALM *realm;
+	{
+		if ((tmp = fr_pair_find_by_num(request->config, PW_PROXY_TO_REALM, 0, TAG_ANY)) != NULL) {
+			REALM *realm;
 
-		realm = realm_find2(tmp->vp_strvalue);
+			realm = realm_find2(tmp->vp_strvalue);
 
-		/*
-		 *	Don't authenticate, as the request is going to
-		 *	be proxied.
-		 */
-		if (realm && realm->auth_pool) {
-			return RLM_MODULE_OK;
-		}
+			/*
+			 *	Don't authenticate, as the request is going to
+			 *	be proxied.
+			 */
+			if (realm && realm->auth_pool) {
+				return RLM_MODULE_OK;
+			}
 
-		/*
-		 *	Catch users who set Proxy-To-Realm to a LOCAL
-		 *	realm (sigh).  But don't complain if it is
-		 *	*the* LOCAL realm.
-		 */
-		if (realm &&(strcmp(realm->name, "LOCAL") != 0)) {
-			RWDEBUG2("You set Proxy-To-Realm = %s, but it is a LOCAL realm!  Cancelling proxy request.", realm->name);
-		}
+			/*
+			 *	Catch users who set Proxy-To-Realm to a LOCAL
+			 *	realm (sigh).  But don't complain if it is
+			 *	*the* LOCAL realm.
+			 */
+			if (realm &&(strcmp(realm->name, "LOCAL") != 0)) {
+				RWDEBUG2("You set Proxy-To-Realm = %s, but it is a LOCAL realm!  Cancelling proxy request.", realm->name);
+			}
 
-		if (!realm) {
-			RWDEBUG2("You set Proxy-To-Realm = %s, but the realm does not exist!  Cancelling invalid proxy request.", tmp->vp_strvalue);
+			if (!realm) {
+				RWDEBUG2("You set Proxy-To-Realm = %s, but the realm does not exist!  Cancelling invalid proxy request.", tmp->vp_strvalue);
+			}
+		} else if (((tmp = fr_pair_find_by_num(request->config, PW_HOME_SERVER_POOL, 0, TAG_ANY)) != NULL) ||
+				   ((tmp = fr_pair_find_by_num(request->config, PW_PACKET_DST_IP_ADDRESS, 0, TAG_ANY)) != NULL) ||
+				   ((tmp = fr_pair_find_by_num(request->config, PW_PACKET_DST_IPV6_ADDRESS, 0, TAG_ANY)) != NULL) ||
+				   ((tmp = fr_pair_find_by_num(request->config, PW_HOME_SERVER_NAME, 0, TAG_ANY)) != NULL)) {
+			RDEBUG("Proxying due to %s = %s", tmp->da->name, tmp->vp_strvalue);
```

Comment:
we can't use `vp->vp_strvalue` for IPv4 or IPv6 attributes.

Reply:
It was a typo. fixed & updated.